File: Frontend/composeApp/src/commonMain/kotlin/com/teamnotfound/airise/data/serializable/HealthData.kt
package com.teamnotfound.airise.data.serializable

import kotlinx.datetime.LocalDate
import kotlinx.serialization.Serializable

@Serializable
data class HealthData(
    val sleep: Double? = null, //hours
    val steps: Int? = null,
    val caloriesBurned: Int? = null,
    val caloriesEaten: Int? = null,
    val caloriesTarget: Int? = null,
    val hydration: Double? = null,
    val hydrationTarget: Double? = null, //ounces
    var localDate: LocalDate? = null
)

File: Backend/Model/DTOs/HealthData.cs
using System.Text.Json.Serialization;

namespace AiRise.Models.DTOs
{
    public class HealthData
    {
        [JsonPropertyName("caloriesBurned")]
        public int? CaloriesBurned { get; set; }

        [JsonPropertyName("caloriesEaten")]
        public int? CaloriesEaten { get; set; }

        [JsonPropertyName("caloriesTarget")]
        public int? CaloriesTarget { get; set; }

        [JsonPropertyName("steps")]
        public int? Steps { get; set; }

        [JsonPropertyName("sleep")]
        public double? Sleep { get; set; }

        [JsonPropertyName("hydration")]
        public double? Hydration { get; set; }

        [JsonPropertyName("hydrationTarget")]
        public double? HydrationTarget { get; set; }

        [JsonPropertyName("localDate")]
        public DateOnly LocalDate { get; set; } // Format: "YYYY-MM-DD"
    }
}